mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -0,0 +1,161 @@
+      if (hmsTable.getParameters().get(InputFormatConfig.HIVE_DELETE_BACKING_TABLE) == null) {
+        hmsTable.getParameters().put(InputFormatConfig.HIVE_DELETE_BACKING_TABLE, "TRUE");
+      }
+
+      // Set the table type even for non HiveCatalog based tables
+      hmsTable.getParameters().put(BaseMetastoreTableOperations.TABLE_TYPE_PROP,
+          BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toUpperCase());
+
+      // Remove creation related properties
+      PARAMETERS_TO_REMOVE.forEach(hmsTable.getParameters()::remove);
+    }
+  }
+
+  @Override
+  public void rollbackCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    // do nothing
+  }
+
+  @Override
+  public void commitCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    if (icebergTable == null) {
+      catalogProperties.put(HiveTableOperations.TABLE_FROM_HIVE, true);
+      LOG.info("Iceberg table creation with the following properties {}", catalogProperties.keySet());
+      Catalogs.createTable(conf, catalogProperties);
+    }
+  }
+
+  @Override
+  public void preDropTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) throws MetaException {
+    catalogProperties = getCatalogProperties(hmsTable);
+    deleteIcebergTable = hmsTable.getParameters() != null &&
+        "TRUE".equalsIgnoreCase(hmsTable.getParameters().get(InputFormatConfig.HIVE_DELETE_BACKING_TABLE));
+
+    if (!deleteIcebergTable) {
+      if (!Catalogs.canWorkWithoutHive(conf)) {
+        // This should happen only if someone were manually removing this property from the table, or
+        // added the table from outside of Hive
+        throw new MetaException("Can not drop Hive table and keep Iceberg table data when using HiveCatalog. " +
+            "Please add " + InputFormatConfig.HIVE_DELETE_BACKING_TABLE + "='TRUE' to TBLPROPERTIES " +
+            "of the Hive table to enable dropping");
+      }
+    }
+  }
+
+  @Override
+  public void rollbackDropTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    // do nothing
+  }
+
+  @Override
+  public void commitDropTable(org.apache.hadoop.hive.metastore.api.Table hmsTable, boolean deleteData) {
+    if (deleteData && deleteIcebergTable) {
+      LOG.info("Dropping with purge all the data for table {}.{}", hmsTable.getDbName(), hmsTable.getTableName());
+      Catalogs.dropTable(conf, catalogProperties);
+    }
+  }
+
+  private Properties getCatalogProperties(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    Properties properties = new Properties();
+    properties.putAll(hmsTable.getParameters());
+
+    if (properties.get(Catalogs.LOCATION) == null &&
+        hmsTable.getSd() != null && hmsTable.getSd().getLocation() != null) {
+      properties.put(Catalogs.LOCATION, hmsTable.getSd().getLocation());
+    }
+
+    if (properties.get(Catalogs.NAME) == null) {
+      properties.put(Catalogs.NAME, hmsTable.getDbName() + "." + hmsTable.getTableName());
+    }
```

Comment:
I think what is happening here should be documented clearly in the code.

Reply:
Added a comment, and modified a bit to use TableIdentifier instead of concatenating the strings